types/react-native/test/index.tsx

```diff
@@ -195,6 +195,74 @@ const s = StyleSheet.create({
 });
 const f1: TextStyle = s.shouldWork;
 
+// StyleSheet.compose
+// It creates a new style object by composing two existing styles
+const composeTextStyle: StyleProp<TextStyle> = {
+    color: '#000000',
+    fontSize: 20,
+};
+
+const composeImageStyle: StyleProp<ImageStyle> = {
+    resizeMode: 'contain',
+};
+
+// The following use of the compose method is valid
+const combinedStyle = StyleSheet.compose(
+    composeTextStyle,
+    composeTextStyle,
+);
+
+const combinedStyle1 = StyleSheet.compose(
+    composeImageStyle,
+    composeImageStyle,
+);
+
+const combinedStyle2 = StyleSheet.compose(
+    [composeTextStyle],
+    [composeTextStyle],
+);
+
+const combinedStyle3 = StyleSheet.compose(
+    composeTextStyle,
+    null,
+);
+
+const combinedStyle4 = StyleSheet.compose(
+    [composeTextStyle],
+    null,
+);
+
+const combinedStyle5 = StyleSheet.compose(
+    composeTextStyle,
+    Math.random() < 0.5 ? composeTextStyle : null,
+);
+
+const combinedStyle6 = StyleSheet.compose(
+    null,
+    null,
+);
+
+// The following use of the compose method is invalid:
```

Comment:
You can uncomment these and add a `$ExpectError` comment above them to have the test assertion actually being verified.

Reply:
Aha, thanks, will change it.

About `?` - it means an optional parameter or it just simply adds ` | undefined` to the type.

I have used it in the previous commit.

If you will go to the StyleProp<T> type definition, you will see that it includes `Falsy` type.

The `Falsy` type itself has the following definition - `type Falsy = undefined | null | false;` so the `?` doesn't make sense.

Reply:
Re `?`, I see, I was just looking at the complete diff. Looking at [the Flow typing](https://github.com/facebook/react-native/blob/master/Libraries/StyleSheet/StyleSheet.js#L265-L266) of the API, the current version of the PR does look correct.

Just one note, in case there was some misunderstanding there, but wether or not `Falsy` includes `undefined` doesn’t matter in regards to `?`, as that is meant to signify an optional parameter, which just happens to become `undefined`.

```ts
// Required parameter that may be undefined
function foo(x: string | undefined) {}

// Optional parameter
function foo(x?: string) {}
```

Reply:
Done!